File: muchos-motors-front/src/app/shared/models/OrderItem.ts
import { CarPart } from './CarPart';

export class OrderItem {
  constructor(carPart: CarPart) {
    this.carPart = carPart;
    this.carPartId = carPart.carPartId;
  }

  carPart: CarPart;
  carPartId: number | undefined;
  quantity: number = 1;
  price!: number;

  calculatePrice(): number {
    this.price = this.carPart.sellingPrice * this.quantity;
    return this.price;
  }
}

File: muchos-motors/muchos-motors-api/Models/OrderItem.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace muchos_motors_api.Models
{
    public class OrderItem
    {
        [Key]
        public int OrderItemId { get; set; }
        public int OrderId { get; set; }
        [ForeignKey("OrderId")]
        public Order Order { get; set; }

        public int CarPartId { get; set; }
        [ForeignKey("CarPartId")]
        public CarPart CarPart { get; set; }

        public int Quantity { get; set; }
        [Column(TypeName = "decimal(18,2)")]
        public decimal Price { get; set; }
        public bool IsValid { get; set; }


    }
}
